### PR TITLE
Convert comments to puppet-strings

### PR DESCRIPTION
**Pull request by bastelfreak**
With the help of perl!

```perl
use strict;
use warnings;
use File::Glob ':bsd_glob';
my @files = bsd_glob('manifests/*.pp');

foreach my $file (@files) {
  print $file;
  open(my $fh, '<:encoding(UTF-8)', $file)
    or die "Could not open file '$file' $!";

  my @file_content_as_array;
  while (my $row = <$fh>) {
    #chomp $row;
    if ($row =~ /\[\*(.*)\*\]/) {
      push @file_content_as_array, "# \@param ${1}\n";
    } else {
      push @file_content_as_array, "$row";
    }
  }

  close $fh or die "can't close $file: $!";

  print @file_content_as_array;

  open(my $new_fh, '>:encoding(UTF-8)', $file)
    or die "Could not open file '$file' $!";

  foreach (@file_content_as_array) {
    print $new_fh "$_"; # Print each entry in our array to the file
  }

  close $new_fh or die "can't close $file: $!";
}
```

not pretty, but worked well. I did a bit of manual cleanup. We had a few
classes that had documentation for non-existant parameters. I deleted
those.

### CODE REVIEW

File: manifests/server.pp [hunk 1]
```diff
@@ -1,5 +1,6 @@
-# class to manage the actual prometheus server
-# this is a private class that gets called from the init.pp
+#
+# @summary class to manage the actual prometheus server. This is a private class that gets called from the init.pp
```
Comment (alexjfisher) [inline manifests/server.pp]: This class can't actually decide if it's private or not.  README says no, but then there are actually quite a few things you can't set if you don't do it via the main class.

File: manifests/service_reload.pp [hunk 2]
```diff
@@ -1,6 +1,8 @@
-# This class implements prometheus service reload
-# without restarting the whole service when a config
-# changes
+#
+# @summary
+#   This class implements prometheus service reload without
+#   restarting the whole service when a config changes
+#
```
Comment (alexjfisher) [inline manifests/service_reload.pp]: I think this class *is* definitely `@api private` though.

Comment (bastelfreak) [reply manifests/service_reload.pp]: fixed it

File: manifests/server.pp (hunk 1 above)
Comment (bastelfreak) [reply manifests/server.pp]: correct, I will remove the `private` statement

**Comment by alexjfisher** [comment]
This looks like a good first step to improving the docs.  Ideally, I think the parameters that aren't OS specific could come out of hiera so we could let puppet strings pick them up automatically.

And maybe we should figure out how to reduce all the duplication by just linking certain exporter parameter docs to the daemon docs or something??

But there's nothing wrong with getting this merged meanwhile I don' think.

**Comment by bastelfreak** [comment]
@alexjfisher rebased :)
After this is merged we can review hiera existing params and move data from hiera to the exporters.

File: manifests/alertmanager.pp [hunk 3]
```diff
@@ -1,47 +1,45 @@
-# Class: prometheus::alertmanager
 #
-# This module manages prometheus alertmanager
+# @summary This module manages prometheus alertmanager
 #
-# Parameters:
-#  [*arch*]
+# @param arch
 #  Architecture (amd64 or i386)
```
Comment (ekohl) [inline manifests/alertmanager.pp:3]: I think I've seen the need to correctly indent it with 3 spaces or it wasn't picked up. Not sure if that's also the case here.
```suggestion
#   Architecture (amd64 or i386)
```

Comment (bastelfreak) [reply manifests/alertmanager.pp:3]: If I read the REFERENCE.md correctly the current indentation works fine.

**Review by cy4n — APPROVED** [review]

**Pull request merged**